tests/resources/test_file_link.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+from uuid import uuid4
+from mock import patch, Mock
+from botocore.exceptions import ClientError
+
+from citrine.resources.file_link import FileCollection, FileLink, _Uploader
+from tests.utils.session import FakeSession
+from tests.utils.factories import FileLinkDataFactory
+
+
+@pytest.fixture
+def session() -> FakeSession:
+    return FakeSession()
+
+
+@pytest.fixture
+def collection(session) -> FileCollection:
+    return FileCollection(
+        project_id=uuid4(),
+        dataset_id=uuid4(),
+        session=session
+    )
+
+
+@pytest.fixture
+def valid_data() -> dict:
+    return FileLinkDataFactory(url='www.citrine.io', filename='materials.txt')
+
+
+def test_build_equivalence(collection, valid_data):
+    """Test that build() works the same whether called from FileLink or FileCollection."""
+    assert collection.build(valid_data).dump() == FileLink.build(valid_data).dump()
+
+
+def test_string_representation(valid_data):
+    """Test the string representation."""
+    assert str(FileLink.build(valid_data)) == '<File link \'materials.txt\'>'
+
+
+@pytest.fixture
+def uploader() -> _Uploader:
+    """An _Uploader object with all of its fields filled in."""
+    uploader = _Uploader()
+    uploader.bucket = 'citrine-datasvc'
+    uploader.object_key = '334455'
+    uploader.upload_id = 'dea3a-555'
+    uploader.region_name = 'us-west'
+    uploader.aws_access_key_id = 'sahgkjsgahnei'
+    uploader.aws_secret_access_key = 'kdydahkd78452978'
+    uploader.aws_session_token = 'sdfhuaf74yf783g4ofg7g3o'
+    uploader.object_key = '234787521--abcde'
+    uploader.s3_version = '2'
+    return uploader
+
+
+class MockClient(object):
```

Comment:
We should be explicit about naming here.  https://martinfowler.com/articles/mocksArentStubs.html
I would suggest either `FakeS3Client` or `StubS3Client` - test doubles come in different flavors.

Also, moving to `utils` with the other test doubles might organize things a bit more.

Reply:
Done